api/src/main/java/org/motechproject/nms/api/web/UserController.java

```diff
@@ -114,16 +113,24 @@ public UserResponse getUserDetails(@PathVariable String serviceName,
         return user;
     }
 
-    private UserResponse getKilkariResponseUser(Long callingNumber) {
+    private UserResponse getKilkariResponseUser(Long callingNumber, String circle) {
         KilkariUserResponse user = new KilkariUserResponse();
-        Subscriber subscriber = kilkariService.getSubscriber(callingNumber);
-        if (subscriber == null) {
-            throw new NotFoundException(String.format(NOT_FOUND, CALLING_NUMBER));
-        }
-        Set<Subscription> subscriptions = subscriber.getSubscriptions();
         Set<String> packs = new HashSet<>();
-        for (Subscription subscription : subscriptions) {
-            packs.add(subscription.getSubscriptionPack().getName());
+        Subscriber subscriber = kilkariService.getSubscriber(callingNumber);
+        if (subscriber != null) {
+            Set<Subscription> subscriptions = subscriber.getSubscriptions();
+            for (Subscription subscription : subscriptions) {
+                packs.add(subscription.getSubscriptionPack().getName());
+            }
+        } else {
+            /*
+            No subscriber found in the database, according to https://github.com/koshalt/mim/issues/9#event-288912397
+            we need to respond with an empty subscription list, the circle provider in the request and our guess as
+            to what the defaultLanguageLocationCode is, which is done after this method returns
+            */
+            user.setCircle(circle);
+            //todo: figure out the defaultLanguageLocationCode, should that be coming from the language module?
+            user.setDefaultLanguageLocationCode("??");
```

Comment:
Is this if block needed?  According to issue #26 We aren't returning circle anymore.  Also according to issue #27 We are combining the return for new and existing.  So just send back both languageLocationCode and defaultLanguageLocationCode.  Also you can find the default language the same we we find it in MA/MK (https://github.com/motech-implementations/mim/blob/master/api/src/main/java/org/motechproject/nms/api/web/UserController.java#L109)